railties/lib/rails/application/configuration.rb

```diff
@@ -237,6 +237,12 @@ def load_defaults(target_version)
 
           if respond_to?(:action_controller)
             action_controller.raise_on_open_redirects = true
+
+            # This can't use the standard configuration pattern because you can call
+            # wrap_parameters at the top level or on any controller.
+            ActiveSupport.on_load(:action_controller) do
+              ActionController::Base.wrap_parameters format: [:json]
```

Comment:
You broke `ActionController::Api` by classing `ActionController::Base` explicitly. Also fixing in a follow-up commit.